### PR TITLE
chore: restructures and refactors test setup

### LINKED ISSUE #15: Upgrade Mocha to 10.x
Project is a bit overdue for a Mocha upgrade to take advantage of some of the newer features in versions like global fixtures and newer node support.

https://github.com/mochajs/mocha/releases

Would also get rid of a bunch of deprecation warnings.

### PR DESCRIPTION
**Pull request by osis**
**Short description of work done**
This concludes a deep dive into the test framework that started with #15. Ultimately made the decision to remove the hybrid mocha/jest setup and replace with an updated jest setup to keep the runner consistent with other libs for now.

Quick highlights are are...

### PR Checklist

- [x] I have run linter locally
- [x] I have run unit and integration tests locally
- [x] Update configuration the newest version (readme and const)
- [x] Rebased to master branch / merged master

#### Test dapp
- organizes all assets to the `test/dapp` folder
- logically organized like a typical web project. This initial groundwork will allow this to be a separate package to help isolated dependencies.
- adds `stop()` to help with test teardown
- changes compiled contract JS output to sit with other public files to clarify code intent

#### Test Suite
- Removes mocha, older jest dependencies, and unused config/env files
- Converts mocha assertions to jest
- proper global setup and teardown
- tests separated into wallet internal and dapp interaction tests

#### dAppwright
- fixes transaction options flow for Metamask
- priority is non-optional for transaction options since that is mandatory in Metamask now.
- remove waits from tests and implements in actions to allow extension state to settle before continuing

Closes #15

### CODE REVIEW

File: test/dapp/public/main.js [hunk 1]
```diff
@@ -1,56 +1,60 @@
...
+    signed.textContent = 'signed';
     document.body.appendChild(signed);
   });
 
   const transferButton = document.querySelector('.transfer-button');
   transferButton.addEventListener('click', async function () {
     const accounts = await provider.send('eth_requestAccounts', []);
-    await counterContract.methods.increase().send();
-    await web3.eth.sendTransaction({ to: accounts[0], from: accounts[0], value: web3.utils.toWei('0.01') });
+    // await counterContract.increase().send();
```
Comment (RayzorFlash) [inline test/dapp/public/main.js:50]: comment can be removed

File: test/dapp.spec.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  describe('when confirming a transaction', () => {
+    it('should be able to confirm without altering gas settings', async () => {
+      await testPage.click('.increase-button');
+      await wallet.confirmTransaction();
+
+      // confirms transasction sent successfully
+      await testPage.waitForSelector('#increased');
+    });
+
+    it('should be able to confirm with custom gas settings', async () => {
```
Comment (RayzorFlash) [inline test/dapp.spec.ts:41]: should there be another test that uses the `increase-fees` button?

File: test/wallet.spec.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  });
+
+  it('should lock and unlock', async () => {
+    await wallet.lock();
+    await wallet.unlock('password1234');
+  });
+
+  describe('when adding a network', () => {
+    afterAll(async () => {
+      await wallet.switchNetwork('local');
```
Comment (RayzorFlash) [inline test/wallet.spec.ts:24]: is `local` a valid network for it to switch to?

**Review by RayzorFlash — APPROVED** [review]
The tests could still use a bunch more cleanup/rewrite time, but the overall structural changes LGTM

File: test/dapp/public/data.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+const ContractInfo = {
```
Comment (lavens) [inline test/dapp/public/data.js:1]: wut dis?

File: test/dapp/public/main.js [hunk 5]
```diff
@@ -1,56 +1,60 @@
...
+    await ethereum.request({
+      method: 'eth_sendTransaction',
+      params: [{ to: accounts[0], from: accounts[0], value: '10000000000000000' }],
```
Comment (lavens) [inline test/dapp/public/main.js:53]: sending from and to the same account eh, sus...

File: test/dapp.spec.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    await testPage.waitForSelector('#signed');
+  });
+
+  describe('when confirming a transaction', () => {
+    it('should be able to confirm without altering gas settings', async () => {
+      await testPage.click('.increase-button');
+      await wallet.confirmTransaction();
+
+      // confirms transasction sent successfully
```
[the remark this answers is not in the document]
Comment (osis) [reply test/dapp.spec.ts:37]: fixed in https://github.com/TenKeyLabs/dappwright/pull/29

File: test/dapp.spec.ts [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    it('should be able to confirm with custom gas settings', async () => {
+      await testPage.click('.transfer-button');
+
+      await wallet.confirmTransaction({
+        gas: 21000,
+        priority: 2,
+        gasLimit: 202020,
+      });
+
+      // confirms transasction sent successfully
```
[the remark this answers is not in the document]
Comment (osis) [reply test/dapp.spec.ts:50]: fixed in https://github.com/TenKeyLabs/dappwright/pull/29

File: test/dapp/public/main.js (hunk 1 above)
Comment (osis) [reply test/dapp/public/main.js:50]: this should fixed in https://github.com/TenKeyLabs/dappwright/pull/29 as well

File: test/dapp.spec.ts (hunk 2 above)
Comment (osis) [reply test/dapp.spec.ts:41]: that's dormant functionality I think. Will create an issue to investigate.

File: test/wallet.spec.ts (hunk 3 above)
Comment (osis) [reply test/wallet.spec.ts:24]: no, but part of the test is to validate partial matches

File: test/dapp/public/main.js (hunk 5 above)
Comment (osis) [reply test/dapp/public/main.js:53]: haha it is indeed. This will be part of the same investigate mentioned above.

File: test/dapp.spec.ts (hunk 2 above)
Comment (MSALEEM0777) [reply test/dapp.spec.ts:41]: []()

File: test/dapp.spec.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (MSALEEM0777) [reply test/dapp.spec.ts:37]: @msaleem0777

[the remark this answers is not in the document]
Comment (MSALEEM0777) [reply test/dapp.spec.ts:37]: Hi

[the remark this answers is not in the document]
Comment (MSALEEM0777) [reply test/dapp.spec.ts:37]: __

**Pull request merged**